### ISSUE TITLE
executor @nx/js:node source mapping broken

### ISSUE THREAD

**Issue by SamuelColacchia** [nrwl/nx#17258]
### Current Behavior

When running any project that uses "executor": "@nx/js:node" sourcemapping does not work.

### Expected Behavior

Source mapping should work if enabled in project.json and tsconfig

### Steps to Reproduce

1. Create a blank nx project in my case im using `npx create-nx-workspace my-workspace --preset=nest`
2. Produce something where sourcemapping would be used, for example 
``` TS
import { Controller, Get } from "@nestjs/common";

import { AppService } from "./app.service";

@Controller()
export class AppController {
  constructor(private readonly appService: AppService) {}

  @Get()
  getData() {
    throw new Error('error')
  }
  
}
```
3. Observe in the console that the error reports `main.js` rather than `app.controller.js`

``` Shell
[Nest] 124829  - 05/26/2023, 3:00:51 PM   ERROR [ExceptionsHandler] error

Error: error
    at AppController.getData (/home/username/vas/dist/apps/api/main.js:95:15)
    at /home/username/vas/node_modules/@nestjs/core/router/router-execution-context.js:38:29
    at InterceptorsConsumer.intercept (/home/username/vas/node_modules/@nestjs/core/interceptors/interceptors-consumer.js:11:20)
    at /home/username/vas/node_modules/@nestjs/core/router/router-execution-context.js:46:60
    at /home/username/vas/node_modules/@nestjs/core/router/router-proxy.js:9:23
    at Layer.handle [as handle_request] (/home/username/vas/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/username/vas/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/home/username/vas/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/home/username/vas/node_modules/express/lib/router/layer.js:95:5)
    at /home/username/vas/node_modules/express/lib/router/index.js:284:15
```

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 18.16.0
   OS     : linux x64
   npm    : 9.6.6
   Hasher : Native

   nx                 : 16.3.0-beta.11
   @nx/js             : 16.3.0-beta.11
   @nx/jest           : 16.3.0-beta.11
   @nx/linter         : 16.3.0-beta.11
   @nx/workspace      : 16.3.0-beta.11
   @nx/angular        : 16.3.0-beta.11
   @nx/cypress        : 16.3.0-beta.11
   @nx/devkit         : 16.3.0-beta.11
   @nx/eslint-plugin  : 16.3.0-beta.11
   @nx/nest           : 16.3.0-beta.11
   @nx/node           : 16.3.0-beta.11
   @nrwl/tao          : 16.3.0-beta.11
   @nx/web            : 16.3.0-beta.11
   @nx/webpack        : 16.3.0-beta.11
   typescript         : 5.0.4
   ---------------------------------------
   Community plugins:
   apollo-angular : 4.2.1
   ---------------------------------------
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows - WSL2
- [ ] Other (Please specify)

### Additional Information

Doing some debuging of the arguments passed to the node executable i noticed the input args to node js to be.
``` bash
'/home/username/.nvm/versions/node/v18.16.0/bin/node',
'/home/username/vas/node_modules/@nx/js/src/executors/node/node-with-require-overrides',
'-r',
'/home/username/vas/node_modules/@nx/js/node_modules/source-map-support/register.js',
'--inspect=localhost:9229'
```
Exploring and reading through `node-with-require-overrides` it appears that the order is wrong.

From the command line if i set `NX_MAPPINGS` and `NX_FILE_TO_RUN` and then run a similar command to the one below (with the current username) source mapping works !!!!

``` terminal
/home/username/.nvm/versions/node/v18.16.0/bin/node 
-r /home/username/vas/node_modules/@nx/js/node_modules/source-map-support/register.js
/home/username/vas/node_modules/@nx/js/src/executors/node/node-with-require-overrides
```

```
[Nest] 124986  - 05/26/2023, 3:03:57 PM   ERROR [ExceptionsHandler] error
Error: error
    at AppController.getData (/home/username/vas/dist/apps/api/webpack:/@vas/api/src/app/app.controller.ts:13:1)
    at /home/username/vas/node_modules/@nestjs/core/router/router-execution-context.js:38:29
    at InterceptorsConsumer.intercept (/home/username/vas/node_modules/@nestjs/core/interceptors/interceptors-consumer.js:11:20)
    at /home/username/vas/node_modules/@nestjs/core/router/router-execution-context.js:46:60
    at /home/username/vas/node_modules/@nestjs/core/router/router-proxy.js:9:23
    at Layer.handle [as handle_request] (/home/username/vas/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/username/vas/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/home/username/vas/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/home/username/vas/node_modules/express/lib/router/layer.js:95:5)
    at /home/username/vas/node_modules/express/lib/router/index.js:284:15
```

I am not very familiar with this project does any one have a idea where this change should be implemented ?

**Comment by AgentEnder** [comment]
Hey @SamuelColacchia! Thanks for the additional information there at the end, it was super helpful. You can check the PR if you are interested in learning more about how the codebase is structured.

**Comment by SamuelColacchia** [comment]
@AgentEnder That was a very quick reply and change.

Reading over your pull request and applying those changes on my end it fixed the issue. **Additionally it also fixed the debugger!**

**Comment by pmosconi** [comment]
Hi @AgentEnder , is this related to https://github.com/nrwl/nx/issues/14708#issuecomment-1565319863 ?
Thanks

**Comment by AgentEnder** [comment]
I'm not sure

**Comment by SamuelColacchia** [comment]
@pmosconi I dont think this solves that issue will fix #14708 though it should fix your issue since the bug is present in `16.2.2`  [link to code with error](https://github.com/nrwl/nx/blob/1577131e33ec7355ac7b5ab1b401ed55a69fcc98/packages/js/src/executors/node/node.impl.ts#L138)
